x-pack/plugins/security_solution/common/endpoint/service/authz/authz.ts

```diff
@@ -46,130 +58,160 @@ function hasPermission(
   );
   const canReadTrustedApplications =
     canWriteTrustedApplications ||
-    hasPermission(
+    hasKibanaPrivilege(
       fleetAuthz,
       isEndpointRbacEnabled,
       hasEndpointManagementAccess,
       'readTrustedApplications'
     );
-  const canWriteHostIsolationExceptions = hasPermission(
+  const hasWriteHostIsolationExceptionsPermission = hasKibanaPrivilege(
     fleetAuthz,
     isEndpointRbacEnabled,
     hasEndpointManagementAccess,
     'writeHostIsolationExceptions'
   );
-  const canReadHostIsolationExceptions =
-    canWriteHostIsolationExceptions ||
-    hasPermission(
+  const hasReadHostIsolationExceptionsPermission =
+    hasWriteHostIsolationExceptionsPermission ||
+    hasKibanaPrivilege(
       fleetAuthz,
       isEndpointRbacEnabled,
       hasEndpointManagementAccess,
       'readHostIsolationExceptions'
     );
-  const canWriteBlocklist = hasPermission(
+  const canWriteBlocklist = hasKibanaPrivilege(
     fleetAuthz,
     isEndpointRbacEnabled,
     hasEndpointManagementAccess,
     'writeBlocklist'
   );
   const canReadBlocklist =
     canWriteBlocklist ||
-    hasPermission(fleetAuthz, isEndpointRbacEnabled, hasEndpointManagementAccess, 'readBlocklist');
-  const canWriteEventFilters = hasPermission(
+    hasKibanaPrivilege(
+      fleetAuthz,
+      isEndpointRbacEnabled,
+      hasEndpointManagementAccess,
+      'readBlocklist'
+    );
+  const canWriteEventFilters = hasKibanaPrivilege(
     fleetAuthz,
     isEndpointRbacEnabled,
     hasEndpointManagementAccess,
     'writeEventFilters'
   );
   const canReadEventFilters =
     canWriteEventFilters ||
-    hasPermission(
+    hasKibanaPrivilege(
       fleetAuthz,
       isEndpointRbacEnabled,
       hasEndpointManagementAccess,
       'readEventFilters'
     );
-  const canWriteFileOperations = hasPermission(
+  const canWriteFileOperations = hasKibanaPrivilege(
     fleetAuthz,
     isEndpointRbacEnabled,
     hasEndpointManagementAccess,
     'writeFileOperations'
   );
 
+  // Calculate the Host Isolation Exceptions Authz. Some of these authz properties could be
+  // set to `true` in cases where license was downgraded, but entries still exist.
+  const canWriteHostIsolationExceptions =
+    hasWriteHostIsolationExceptionsPermission && isPlatinumPlusLicense;
+
+  const canReadHostIsolationExceptions =
+    canWriteHostIsolationExceptions ||
+    (hasReadHostIsolationExceptionsPermission && isPlatinumPlusLicense) ||
+    // Should be able to see Host Isolation Exceptions if license was downgraded
+    (hasReadHostIsolationExceptionsPermission &&
+      !isPlatinumPlusLicense &&
+      hasHostIsolationExceptionsItems);
+
+  const canDeleteHostIsolationExceptions =
+    canWriteHostIsolationExceptions ||
+    // Should be able to delete if host isolation exceptions exists and license is not platinum+
+    (hasWriteHostIsolationExceptionsPermission &&
+      !isPlatinumPlusLicense &&
+      hasHostIsolationExceptionsItems);
```

Comment:
Does this cover the case when HIE RBAC is set to `None` or `Read` (regardless of the license) but `hasHostIsolationExceptionsItems` is `true`? We should allow the deletion of artifacts in that case as well. Right?

Reply:
Delete is tied to the `write` privilege. Only users with this privilege should be able to delete entries.

Reply:
@ashokaditya -  @dasansol92 is correct. The ability to Delete still needs to be tied to the RBAC privilege of `write` - else you would be giving authz. to a user that potentially did not have permission to CRUD (`all`).